lib/isafw/isaplugins/ISA_cfa_plugin.py

```diff
@@ -62,44 +61,38 @@ def __init__(self, ISA_config):
-            if rc == 0:
-                # check that execstack is installed
-                rc = subprocess.call(["which", "readelf"], stdout=DEVNULL, stderr=DEVNULL)
-                if rc == 0:
-                    self.initialized = True
-                    with open(self.logfile, 'w') as flog:
-                        flog.write("\nPlugin ISA_CFChecker initialized!\n")
-                        DEVNULL.close()
-                    return
+        self.ISA_filesystem = ""
+        # check that checksec and other tools are installed
+        tools_errors = _check_tools()
+        if tools_errors:
+            with open(self.logfile, 'w') as flog:
+                flog.write(tools_errors)
+                return
+        self.initialized = True
         with open(self.logfile, 'w') as flog:
-            flog.write("checksec, execstack or readelf tools are missing!\n")
-            flog.write("Please install checksec from http://www.trapkit.de/tools/checksec.html\n")
-            flog.write("Please install execstack from prelink package\n")
-        DEVNULL.close()
+            flog.write("\nPlugin ISA_CFChecker initialized!\n")
+        return
 
     def process_filesystem(self, ISA_filesystem):
+        self.ISA_filesystem = ISA_filesystem
+        fs_path = self.ISA_filesystem.path_to_fs
+        img_name = self.ISA_filesystem.img_name
         if (self.initialized):
-            if (ISA_filesystem.img_name and ISA_filesystem.path_to_fs):
+            if (img_name and fs_path):
                 with open(self.logfile, 'a') as flog:
-                    flog.write("\n\nFilesystem path is: " + ISA_filesystem.path_to_fs)
+                    flog.write("\n\nFilesystem path is: " + fs_path)
                 if self.full_reports:
-                    with open(self.full_report_name + "_" + ISA_filesystem.img_name, 'w') as ffull_report:
+                    with open(self.full_report_name + "_" + img_name, 'w') as ffull_report:
                         ffull_report.write(
-                            "Security-relevant flags for executables for image: " + ISA_filesystem.img_name + '\n')
-                        ffull_report.write("With rootfs location at " +
-                                           ISA_filesystem.path_to_fs + "\n\n")
-                self.files = self.find_files(ISA_filesystem.path_to_fs)
-                with open(self.logfile, 'a') as flog:
-                    flog.write("\n\nFile list is: " + str(self.files))
-                self.process_files(ISA_filesystem.img_name, ISA_filesystem.path_to_fs)
-                self.write_report(ISA_filesystem)
-                self.write_report_xml(ISA_filesystem)
+                            "Security-relevant flags for executables for image: " + img_name + '\n')
+                        ffull_report.write("With rootfs location at " + fs_path + "\n\n")
+                files = self.find_files(fs_path)
+                import multiprocessing
```

Comment:
note: in case we wouldn't be successful with forked workers, we can use
```python
from multiprocessing.dummy import Pool
```
to switch to Threaded implementation.